types/row_proof_test.go

```diff
@@ -53,6 +53,12 @@ func TestRowProofValidate(t *testing.T) {
 			root:    incorrectRoot,
 			wantErr: true,
 		},
+		{
+			name:    "start row greater than end row",
+			rp:      RowProof{StartRow: 10, EndRow: 5},
+			root:    root,
+			wantErr: true,
+		},
```

Comment:
[optional] should we add a test case where start row == end row?

Reply:
start == end is a valid case which won't be after a PR that I will be opening subsequently. So decided not to include that case